### ISSUE TITLE
feat: extend codeblock height of shiki blocks

### ISSUE THREAD

**Issue by juliusmarminge** [fuma-nama/fumadocs#234]
**Is your feature request related to a problem? Please describe.**

Looking at your example here: https://fumadocs.vercel.app/docs/ui/twoslash#example
![CleanShot 2024-01-31 at 15 13 13](https://github.com/fuma-nama/fumadocs/assets/51714798/df4a9d27-4688-4cda-8a2d-4c426befa53d)

I don't think that a that small codeblock with scrolling behavior is the best user experience. Would be nice if the height was expanded to fit the shiki block:

![CleanShot 2024-01-31 at 15 16 48](https://github.com/fuma-nama/fumadocs/assets/51714798/4e08493f-78e0-4552-abc4-97b5605c7d5d)

**Describe the solution you'd like**

An option for this to be the default without having to hack into the CSS in userland

**Describe alternatives you've considered**

I'm guessing I could do this in userland by overriding some classses but that feels hacky

**Comment by fuma-nama** [comment]
Fumadocs uses the 1.0.0 beta version of Shiki, which enables rich renderer by default. Rich renderer adds the auto-complete popup to the same span of your token, therefore it is difficult to replicate the same behavior.

The official Shiki site is using empty comments to create similar effects.

```ts
console.lo
//       ^|
          //
          //
          //
```

**Comment by fuma-nama** [comment]
A proper implementation will be released as Shiki 1.0.0 becomes stable

**Comment by fuma-nama** [comment]
Considering the official Shiki site doesn't mention an alternative (instead they've used the same approach above), I'll close this for now.
